hack/charts/cluster-api-operator/values.yaml

```diff
@@ -6,7 +6,17 @@ bootstrap: ""
 controlPlane: ""
 infrastructure: ""
 addon: ""
-manager.featureGates: {}
+manager:
+  featureGates: {}
+# e.g. 
+# manager:
+#   featureGates:
+#     core:
+#       KubeadmBootstrapFormatIgnition: true 
+#     bootstrap:
+#       KubeadmBootstrapFormatIgnition: true 
+#     controlPlane:
+#       KubeadmBootstrapFormatIgnition: true 
```

Comment:
Out of scope for this PR, but `values.yaml` already has core providers listed at the top level - would it make sense to lay-out all the config / params under, instead of duplicating them all over?

e.g.

```yaml
core:            # <- everything for core provider in once place
  version: "v0.1.0"
  manager:
    featureGates:
      KubeadmBootstrapFormatIgnition: false
  deployment:
    ...
  configSecret:
    ...
  fetchConfig:
    ...
bootstrap:       # <- everything for bootstrap provider in once place
  version: ""
  manager:
    featureGates:
      KubeadmBootstrapFormatIgnition: true 
controlPlane:    # <- everything for controlPlane provider in once place
  manager:
    featureGates:
      KubeadmBootstrapFormatIgnition: true  
```

Reply:
I think this approach makes sense. Instead of using key-value pairs, we could use an array. This will provide more flexibility, allowing specific configurations for each provider when multiple providers are used. Let me know what you think, I can work on that. 

```
core:            # <- everything for core provider in once place
- name: "cluster-api"
  version: "v0.1.0"
  manager:
    featureGates:
      KubeadmBootstrapFormatIgnition: false
  deployment:
    ...
  configSecret:
    ...
  fetchConfig:
    ...
bootstrap:
- name: "kubeadm"       # <- everything for bootstrap provider in once place
  version: ""
  manager:
    featureGates:
      KubeadmBootstrapFormatIgnition: true 
controlPlane:
- name: "kubeadm"
  version: ""    # <- everything for controlPlane provider in once place
  manager:
    featureGates:
      KubeadmBootstrapFormatIgnition: true  
```